pkg/lint/rules/chartfile.go

```diff
@@ -104,9 +105,14 @@ func validateChartYamlFormat(chartFileError error) error {
 }
 
 func validateChartName(cf *chart.Metadata) error {
+	var validNameRegex = regexp.MustCompile(`^[a-z0-9](([-a-z0-9]*)?)$`)
```

Comment:
Where did we get this regex from? In part, because we need to ensure it is correct.

Reply:
Only case I think it might miss is when a chart name starts with a "-". But according to the guidelines it should not start (or end) with a "-". Words can be separated by a "-".